newm-server/src/main/kotlin/io/newm/server/features/minting/repo/MintingRepositoryImpl.kt

```diff
@@ -0,0 +1,650 @@
+package io.newm.server.features.minting.repo
+
+import com.google.common.annotations.VisibleForTesting
+import io.newm.chain.grpc.PlutusData
+import io.newm.chain.grpc.plutusData
+import io.newm.chain.grpc.plutusDataList
+import io.newm.chain.grpc.plutusDataMap
+import io.newm.chain.grpc.plutusDataMapItem
+import io.newm.server.features.collaboration.model.Collaboration
+import io.newm.server.features.collaboration.model.CollaborationFilters
+import io.newm.server.features.collaboration.repo.CollaborationRepository
+import io.newm.server.features.song.model.Song
+import io.newm.server.features.user.database.UserEntity
+import io.newm.server.features.user.model.User
+import io.newm.server.features.user.repo.UserRepository
+import io.newm.shared.koin.inject
+import io.newm.txbuilder.ktx.toPlutusData
+import org.jetbrains.exposed.sql.transactions.transaction
+import org.koin.core.parameter.parametersOf
+import org.slf4j.Logger
+import kotlin.time.Duration.Companion.seconds
+
+class MintingRepositoryImpl(
+    private val userRepository: UserRepository,
+    private val collabRepository: CollaborationRepository,
+) : MintingRepository {
+
+    private val log: Logger by inject { parametersOf(javaClass.simpleName) }
+
+    override suspend fun mint(song: Song) {
+        val user = userRepository.get(song.ownerId!!)
+        val collabs = collabRepository.getAll(
+            user.id!!,
+            CollaborationFilters(
+                songIds = listOf(song.id!!),
+                olderThan = null,
+                newerThan = null,
+                ids = null,
+                emails = null
+            ),
+            0,
+            Integer.MAX_VALUE
+        )
+        val metadata = buildStreamTokenMetadata(song, user, collabs)
+
+        TODO("Mint the song on the blockchain.")
```

Comment:
This will be in the next PR. Wanted to keep this one from getting too large so I'm separating the metadata from the actual minting.